minkindr/include/kindr/minimal/implementation/rotation-quaternion-inl.h

```diff
@@ -407,8 +409,9 @@ RotationQuaternionTemplate<Scalar>::operator*(
 
   // check if the multiplication has resulted in the quaternion no longer being
   // approximately normalized.
-  if (std::abs(result.squaredNorm() - static_cast<Scalar>(1.0)) >
-      EPS<Scalar>::normalization_value()) {
```

Comment:
@burrimi I believe @ZacharyTaylor saw the problem also with doubles, as long as huge chains of multiplications were taking place (i.e., optimization loops)